src/posts/2020-07-27-unsucking-your-project-onboarding.md

```diff
@@ -0,0 +1,789 @@
+fails and we `source` the `bin/functions` helpers. We also allow defining an
+`env`, which defaults to `dev` as the first argument.
+
+The first step is then to install the required languages. In this sample
+project, let's assume we are using Elixir for the backend and the frontend is a
+classic React app.
+
+```shell
+pp_info "setup" "Installing required languages..."
+
+if not_installed "asdf"; then
+  pp_error "setup" "
+    We are using asdf (https://github.com/asdf-vm/asdf) to manage tool
+    dependencies, since it was not found on your system we cannot ensure that you
+    are using the correct versions of all the tools. Please install it and run
+    this script again, or proceed at your own peril.
+  "
+
+  ensure_confirmation
+else
+  set +e
+  asdf plugin-add erlang https://github.com/asdf-vm/asdf-erlang.git 2>/dev/null
+  asdf plugin-add elixir https://github.com/asdf-vm/asdf-elixir.git 2>/dev/null
+  asdf plugin-add nodejs https://github.com/asdf-vm/asdf-nodejs.git 2>/dev/null
+  set -e
+
+  printf "Importing nodejs release team keyring... "
+  bin/helpers/import-nodejs-keys
+  echo "Done!"
+
+  asdf install
+fi
+
+pp_info "setup" "resourcing environment"
+source ~/.asdf/asdf.sh
+```
+
+As mentioned, we favour `asdf` for language version management. Usually this
+means we'll commit a `.tool-versions` in the repo to make everyone is running
+the same.
+
+Notice that since `asdf` is not required, we check if it isn't installed and if
+not, we notify the developer and ask for confirmation. Otherwise, we dive right
+into the `asdf` parts.
+
+We run `set +e` before adding the plugins in case it fails since they're
+probably already added. Afterwards we'll need to import the `nodejs` release
+team keyring, before safely installing node, which is conveniently available on
+a `bin/` helper:
+
+```shell
+#!/usr/bin/env sh
+
+set -ue
+
+## Keys from https://github.com/nodejs/node/#release-team
+KEYS="94AE36675C464D64BAFA68DD7434390BDBE9B9C5 \
+      FD3A5288F042B6850C66B31F09FE44734EB7990E \
+      71DCFD284A79C3B38668286BC97EC7A07EDE3FC1 \
+      DD8F2338BAE7501E3DD5AC78C273792F7D83545D \
+      C4F0DFFF4E8C1A8236409D08E73BC641CC11F4C8 \
+      B9AE9905FFD7803F25714661B63B535A4C206CA9 \
+      8FCCA13FEF1D0C2E91008E09770F7A9A5AE15600 \
+      77984A986EBC2AA786BC0F66B01FBB92821C587A \
+      A48C2BEE680E841632CD4E44F07496B3EB3C1762 \
+      4ED778F539E3634C779C87C6D7062848A1AB005C \
+      B9E2F5981AA6E0CD28160D9FF13993A75599653C"
+
+SERVERS="ha.pool.sks-keyservers.net
+         p80.pool.sks-keyservers.net:80 \
+         ipv4.pool.sks-keyservers.net \
+         keyserver.ubuntu.com
+         keyserver.ubuntu.com:80 \
+         pgp.mit.edu
+         pgp.mit.edu:80"
+
+for key in $KEYS; do
+  if ! gpg --list-keys $key 2>&1 >/dev/null; then
+    for server in $SERVERS; do
+      gpg --no-tty --keyserver "hkp://$server" --display-charset utf-8 --recv-keys "$key"  && break
+    done
+  fi
+done
+```
```

Comment:
Why include the script at all? If you want to reference it, wouldn't it be better to just link to its source in the official repository? This is just unnecessary code the reader has to go through for no reason.